test/remote-connector.test.js

```diff
@@ -87,3 +91,55 @@ describe('RemoteConnector', function() {
     });
   });
 });
+
+describe('Custom Path', function() {
+
+  var test = this;
+
+  before(function(done) {
+    
+    var db_config = {
+      connector: loopback.Memory,
+      name: 'db'
+    };
+    var db = loopback.createDataSource('db', db_config);
+
+    var ServerModel_config = {
+      dataSource: 'db',
+      http: {
+        path: '/custom'
+      }
+    };
+
+    var ServerModel = loopback.PersistedModel.extend('TestModel', {foo: 'string'}, ServerModel_config);
+    ServerModel.attachTo(db);
+
+    server = test.server = createAppWithRest();
+    server.model(ServerModel);
+    server.dataSource(db, db_config);
```

Comment:
> I usually do too except in this case the lines were quite long.

Ah, in that case we split arguments to multiple line.

```js
var ServerModel = loopback.PersistedModel.extend('TestModel', {foo: 'string'}, {
  dataSource: 'db',
  http: { path: '/custom'  },
});
// or even
var ServerModel = loopback.PersistedModel.extend(
  'TestModel',
  {foo: 'string'},
  {
    dataSource: 'db',
    http: { path: '/custom'  },
  });
```

Not a big deal though, you can keep your code as it is.

> The "convoluted" setup follows the same pattern as the loopback-boot module. Your simplified proposal does not correctly attach http.path and the test fails.

Ha, good point. So there are two parts mixed in your `ServerModel_config` - model settings applied by "create model" and then model configuration applied by "attach model to an app".

Let me fix my example code to account for that. And BTW the `foo` property is not used anywhere in the test AFAICT, thus it can be removed.

```js
var ServerModel = server.registry.createModel('TestModel', {}, {
  http: { path: '/custom' }
});

server.model(ServerModel, { dataSource: 'db' });
```